### PR TITLE
Enable the duration of the retries to be configurable

### PR DESCRIPTION
**Pull request by tylerturk**
### Reviewers

- [x] @geerlingguy 

cc: 
### High level summary

My vagrant still failed to provision due to Jenkins throwing a 503 after 60 seconds. Let's use the same validation but allow consumers of this role to configure how long they're willing to wait.

:cry: 

### Approach

Extrapolate out the hard-coded timeout and retries to be variables.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+jenkins_connection_delay: 5
+jenkins_connection_retries: 12
```
Comment (geerlingguy) [inline defaults/main.yml]: Let's go back to 300 seconds (so `60` for the default here), just to be generous. The first Jenkins startup can take a while.

File: README.md [hunk 2]
```diff
@@ -27,6 +27,14 @@ The location at which the `jenkins-cli.jar` jarfile will be kept. This is used f
 
 Jenkins plugins to be installed automatically during provisioning. You can always install more plugins via the Jenkins UI at a later time, but this is helpful in getting things up and running more quickly.
 
+    jenkins_connection_delay: 5
+
+The amount of time to wait between connection attempts to the Jenkins web interface to verify it has started.
+
+    jenkins_connection_retries: 12
```
Comment (geerlingguy) [inline README.md]: Let's go back to 300 seconds (so `60` for the default here), just to be generous. The first Jenkins startup can take a while.

File: defaults/main.yml (hunk 1 above)
Comment (tylerturk) [inline defaults/main.yml]: Done!

File: README.md (hunk 2 above)
Comment (tylerturk) [inline README.md]: Done!

**Pull request merged**